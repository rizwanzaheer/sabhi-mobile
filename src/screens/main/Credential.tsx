import React, { useEffect, useState } from 'react'
import {
  Container,
  Credential,
  Device,
  Text,
  RadioBtn,
  Constants,
  Typings,
} from '@kancha/kancha-ui'
import { NavigationStackScreenProps } from 'react-navigation-stack'
import { HeaderButtons, Item } from 'react-navigation-header-buttons'
import { FlatList } from 'react-native'
import { Colors } from '../../theme'
import { ScrollView } from 'react-native-gesture-handler'

interface CredentialStyle {
  background: Typings.BrandPropOptions
  shadow: number
}

interface Props extends NavigationStackScreenProps {}

const CredentialDetail: React.FC<Props> & { sharedElements: any } & {
  navigationOptions: any
} = ({ navigation }) => {
  const credentials = navigation.getParam('credentials')
  const initialCredential = navigation.getParam('credentialIndex', 0)
  const [sharingMode, toggleSharingMode] = useState<boolean>(false)
  const [selected, updateSelected] = useState<number[]>([])

  const isSelected = (index: number) => {
    return selected.includes(index)
  }

  const selectedStyle = (index: number): CredentialStyle => {
    return {
      background: isSelected(index) ? 'primary' : 'primary',
      shadow: 0,
    }
  }

  const selectCredential = (index: number) => {
    if (selected.includes(index)) {
      updateSelected(selected.filter((item: number) => item !== index))
    } else {
      updateSelected(selected.concat([index]))
    }
  }

  const initailOffset = { x: (Device.width - 15) * initialCredential, y: 0 }

  useEffect(() => {
    navigation.setParams({ sharingMode, toggleSharingMode })
  }, [sharingMode])

  const renderCredential = ({
    item,
    index,
  }: {
    item: Typings.VerifiableCredential & {
      raw: string
      issuer: any
      subject: any
      claims: any[]
      expirationDate: any
    }
    index: number
  }) => (
    <Container w={Device.width - 10} padding paddingRight={10}>
      <ScrollView testID={'SCROLLVIEW'} scrollEventThrottle={16}>
        <Credential
          testID={'CREDENTIAL'}
          onPress={() => sharingMode && selectCredential(index)}
          detailMode
          jwt={item.raw}
          issuer={item.issuer}
          subject={item.subject}
          fields={item.claims}
          exp={item.expirationDate}
          {...selectedStyle(index)}
        />
        {sharingMode && (
          <RadioBtn
            testID={'RADIO_BTN'}
            selected={isSelected(index)}
            onPress={() => selectCredential(index)}
          >
            {isSelected(index) && 'Sharing'}
          </RadioBtn>
        )}
      </ScrollView>
    </Container>
  )

  return (
    <Container flex={1} backgroundColor={Colors.BLACK}>
      {sharingMode && (
        <Container padding>
          <Text type={Constants.TextTypes.Body} textColor={Colors.LIGHT_GREY}>
            Select credential(s) for sharing. Flow to be completed.
          </Text>
        </Container>
      )}
      <FlatList
        testID={'FLATLIST'}
        horizontal
        pagingEnabled
        contentOffset={initailOffset}
        showsHorizontalScrollIndicator={false}
        snapToAlignment={'center'}
        keyExtractor={(item: Typings.VerifiableCredential) => item.hash}
        data={credentials}
        renderItem={renderCredential}
      ></FlatList>
    </Container>
  )
}

CredentialDetail.navigationOptions = ({ navigation }: any) => {
  const { sharingMode, toggleSharingMode } = navigation.state.params || {}
  const sharingModeEnabled = navigation.getParam('sharingModeEnabled', true)

  return {
    title: 'Credential',
    headerStyle: {
      backgroundColor: '#f4511e',
    },
    headerLeft: () => (
      <HeaderButtons>
        <Item
          testID={'DONE_BTN'}
          title={'Done'}
          onPress={navigation.dismiss}
          color={Colors.WHITE}
        />
      </HeaderButtons>
    ),
    headerRight: () => (
      <HeaderButtons>
        {sharingMode && sharingModeEnabled ? (
          <Item
            testID={'CANCEL_BTN'}
            title={'Cancel'}
            onPress={() => toggleSharingMode(false)}
            color={Colors.WHITE}
          />
        ) : sharingModeEnabled ? (
          <Item
            testID={'SHARE_BTN'}
            title={'Share'}
            onPress={() => toggleSharingMode(true)}
            color={Colors.WHITE}
          />
        ) : null}
      </HeaderButtons>
    ),
  }
}

CredentialDetail.sharedElements = (navigation: any) => {
  return navigation
    .getParam('credentials')
    .map((vc: Typings.VerifiableCredential) => ({
      id: vc.hash,
      animation: 'fade',
      resize: 'clip',
    }))
}

export default CredentialDetail
