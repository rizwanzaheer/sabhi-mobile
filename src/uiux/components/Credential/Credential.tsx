import * as React from 'react'
import Container from '../Container/Container'
import Text, { TextTypes } from '../Text/Text'
import Avatar from '../Avatar/Avatar'
import Card from '../Card/Card'
import Icon from '../Icon/Icon'
import * as Kancha from '../../types'
import { withTheme } from '../../theming'
import QRCode from 'react-native-qrcode-svg'

const S = require('sugar/string')

interface Field {
  type: string
  value: any
  isObj: boolean
}

export interface CredentialProps {
  onPress?: () => void
  issuer: Kancha.Identity
  subject: Kancha.Identity
  exp: number
  jwt?: string
  qrSize?: number
  fields: Field[]
  testID?: string
  shadow?: number
  background?: Kancha.BrandPropOptions
  detailMode?: boolean
  marginBottom?: number
  theme: any
}

const Credential: React.FC<CredentialProps> = ({
  shadow,
  onPress,
  issuer,
  background,
  fields,
  jwt,
  qrSize,
  subject,
  testID,
  detailMode,
  theme,
  marginBottom,
}) => {
  const subProfileSource = subject.profileImage ? { source: { uri: subject.profileImage } } : {}
  const issProfileSource = issuer.profileImage ? { source: { uri: issuer.profileImage } } : {}
  const credentialFields = fields.filter((field: any, i) => (detailMode ? field : i < 2))

  return (
    <Card
      onPress={onPress}
      testID={testID}
      shadow={shadow || 0}
      background={background}
      marginBottom={marginBottom}
    >
      <Container flexDirection={'row'} alignItems={'flex-start'}>
        <Container>
          <Avatar
            border
            {...subProfileSource}
            address={subject.did}
            type={'circle'}
            gravatarType={'retro'}
            size={40}
          />
          <Container viewStyle={{ position: 'absolute', top: 0, left: 15 }}>
            <Avatar
              border
              {...issProfileSource}
              address={issuer.did}
              type={'circle'}
              gravatarType={'retro'}
              size={40}
            />
          </Container>
        </Container>
        <Container marginLeft={28}>
          <Container>
            <Text bold type={TextTypes.ActivityTitle}>
              {issuer.shortId}
            </Text>
            <Container marginTop={3} flexDirection={'row'} alignItems={'flex-start'}>
              <Container marginRight={5}>
                <Icon
                  icon={{ name: 'ios-play', iconFamily: 'Ionicons' }}
                  size={15}
                  color={theme.colors.primary.text}
                />
              </Container>
              <Text type={TextTypes.ActivityTitle} textStyle={{ fontSize: 15 }}>
                {subject.shortId}
              </Text>
            </Container>
          </Container>
        </Container>
      </Container>
      <Container marginTop={detailMode ? 30 : 16}>
        {credentialFields.map((field: any, i: number) => {
          const fieldValueImage = !field.isObj
            ? field.value.endsWith('.jpg') || field.value.endsWith('.png')
            : false
          return (
            <Container marginBottom={5} key={i}>
              <Container>
                <Text textStyle={{ fontSize: 12, textTransform: 'uppercase' }} type={TextTypes.SubTitle}>
                  {S.String.titleize(field.type)}
                </Text>
              </Container>
              <Container justifyContent={'flex-start'}>
                {fieldValueImage ? (
                  <Container paddingTop={5}>
                    <Avatar
                      source={{ uri: field.value }}
                      address={issuer.did}
                      type={'rounded'}
                      gravatarType={'retro'}
                      size={25}
                    />
                  </Container>
                ) : (
                  <Text type={TextTypes.ActivityTitle}>
                    {field.isObj ? 'Type not supported yet' : field.value}
                  </Text>
                )}
              </Container>
            </Container>
          )
        })}
        {!detailMode && fields.length > 2 && (
          <Container>
            <Text type={TextTypes.SubTitle}>...</Text>
          </Container>
        )}
        {detailMode && jwt && (
          <Container marginTop={50} alignItems={'center'} justifyContent={'center'}>
            <QRCode size={qrSize ? qrSize : 200} value={jwt} />
          </Container>
        )}
      </Container>
    </Card>
  )
}

export default withTheme(Credential)
