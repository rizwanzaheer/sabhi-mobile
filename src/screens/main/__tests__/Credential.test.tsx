import { View, FlatList } from 'react-native'
import React from 'react'
import Credential from '../Credential'
import { createAppContainer, createSwitchNavigator } from 'react-navigation'
import { createStackNavigator } from 'react-navigation-stack'

import {
  render,
  fireEvent,
  waitForElement,
  shallow,
  act,
} from 'react-native-testing-library'

const sertoVerifiableCredential = {
  iss: 'Serto Identity Platform',
  sub: 'did:ethr:0xf3beac30c498d9e26865f34fcaa57dbb935b0d74',
  iat: 1562769371,
  exp: 1579478400,
  hash: 'testHash',
  claim: {
    'Serto ID': {
      name: 'Sarah Adamson',
      dateOfBirth: '22-01-75',
      country: 'USA',
      children: [
        {
          name: 'Bob',
          age: 4,
        },
        {
          name: 'Alice',
          age: 9,
        },
      ],
    },
  },
  vc: [],
}

export const renderWithNavigation = (component: any, options: any) => {
  const Navigator = createAppContainer(
    createStackNavigator({
      CredentialDetail: createStackNavigator({
        Root: () => component,
      }),
    }),
  )
  return render(<Navigator />, options)
}

describe('Credential', () => {
  it('renders correctly with data', async () => {
    const navigation = {
      navigate: jest.fn(),
      getParam: jest
        .fn()
        .mockReturnValue({ ...sertoVerifiableCredential, type: 'Some VC' }),
      setParams: jest.fn(),
      state: {
        params: {
          sharingMode: false,
          toggleSharingMode: jest.fn(),
        },
      },
    }
    //@ts-ignore
    const tree = render(<Credential navigation={navigation} />)

    expect(navigation.getParam).toBeCalled()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly without data', () => {
    const navigation = {
      navigate: jest.fn(),
      getParam: jest.fn().mockReturnValue(null),
      setParams: jest.fn(),
    }
    //@ts-ignore
    const tree = render(<Credential navigation={navigation} />).toJSON()

    expect(navigation.getParam).toBeCalled()
    expect(tree).toMatchSnapshot()
  })

  it('test sharing button', async () => {
    const navigation = {
      navigate: jest.fn(),
      getParam: jest
        .fn()
        .mockReturnValue({ ...sertoVerifiableCredential, type: 'Some VC' }),
      setParams: jest.fn(),
      state: {
        params: {
          sharingMode: false,
          toggleSharingMode: jest.fn(),
        },
      },
    }
    //@ts-ignore
    const tree = render(<Credential navigation={navigation} />)
    const Header = Credential.navigationOptions({ navigation: navigation })
    const headerTree = render(<Header.headerRight />)

    act(() => {
      fireEvent.press(headerTree.getByText('Share'))
    })

    expect(tree.toJSON()).toMatchSnapshot()
    expect(headerTree.toJSON()).toMatchSnapshot()
    expect(navigation.state.params.toggleSharingMode).toHaveBeenCalled()
  })

  it('test cancel sharing', async () => {
    const navigation = {
      navigate: jest.fn(),
      getParam: jest
        .fn()
        .mockReturnValue({ ...sertoVerifiableCredential, type: 'Some VC' }),
      setParams: jest.fn(),
      state: {
        params: {
          sharingMode: true,
          toggleSharingMode: jest.fn(),
          updateSelected: jest.fn(),
        },
      },
    }
    //@ts-ignore
    const tree = render(<Credential navigation={navigation} />)
    const Header = Credential.navigationOptions({ navigation: navigation })
    const headerTree = render(<Header.headerRight />)

    act(() => {
      fireEvent.press(headerTree.getByText('Cancel'))
    })

    expect(tree).toMatchSnapshot()
    expect(headerTree.toJSON()).toMatchSnapshot()
    expect(navigation.state.params.toggleSharingMode).toHaveBeenCalled()
  })

  it('test done button', async () => {
    const navigation = {
      navigate: jest.fn(),
      getParam: jest
        .fn()
        .mockReturnValue({ ...sertoVerifiableCredential, type: 'Some VC' }),
      setParams: jest.fn(),
      dismiss: jest.fn(),
      state: {
        params: {
          sharingMode: false,
          toggleSharingMode: jest.fn(),
        },
      },
    }
    //@ts-ignore
    const tree = render(<Credential navigation={navigation} />)
    const Header = Credential.navigationOptions({ navigation: navigation })
    const headerTree = render(<Header.headerLeft />)
    act(() => {
      fireEvent.press(headerTree.getByText('Done'))
    })

    expect(tree).toMatchSnapshot()
    expect(headerTree.toJSON()).toMatchSnapshot()
    expect(navigation.dismiss).toHaveBeenCalled()
  })
})
