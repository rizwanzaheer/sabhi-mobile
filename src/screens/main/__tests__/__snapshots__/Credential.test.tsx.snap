// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credential renders correctly with data 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#000000",
      "borderRadius": undefined,
      "flex": 1,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <RCTScrollView
    contentOffset={
      Object {
        "x": NaN,
        "y": 0,
      }
    }
    data={
      Object {
        "claim": Object {
          "Serto ID": Object {
            "children": Array [
              Object {
                "age": 4,
                "name": "Bob",
              },
              Object {
                "age": 9,
                "name": "Alice",
              },
            ],
            "country": "USA",
            "dateOfBirth": "22-01-75",
            "name": "Sarah Adamson",
          },
        },
        "exp": 1579478400,
        "hash": "testHash",
        "iat": 1562769371,
        "iss": "Serto Identity Platform",
        "sub": "did:ethr:0xf3beac30c498d9e26865f34fcaa57dbb935b0d74",
        "type": "Some VC",
        "vc": Array [],
      }
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    snapToAlignment="center"
    stickyHeaderIndices={Array []}
    testID="FLATLIST"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`Credential renders correctly without data 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#000000",
      "borderRadius": undefined,
      "flex": 1,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <RCTScrollView
    contentOffset={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    data={null}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    snapToAlignment="center"
    stickyHeaderIndices={Array []}
    testID="FLATLIST"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`Credential test cancel sharing 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#000000",
      "borderRadius": undefined,
      "flex": 1,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <RCTScrollView
    contentOffset={
      Object {
        "x": NaN,
        "y": 0,
      }
    }
    data={
      Object {
        "claim": Object {
          "Serto ID": Object {
            "children": Array [
              Object {
                "age": 4,
                "name": "Bob",
              },
              Object {
                "age": 9,
                "name": "Alice",
              },
            ],
            "country": "USA",
            "dateOfBirth": "22-01-75",
            "name": "Sarah Adamson",
          },
        },
        "exp": 1579478400,
        "hash": "testHash",
        "iat": 1562769371,
        "iss": "Serto Identity Platform",
        "sub": "did:ethr:0xf3beac30c498d9e26865f34fcaa57dbb935b0d74",
        "type": "Some VC",
        "vc": Array [],
      }
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    snapToAlignment="center"
    stickyHeaderIndices={Array []}
    testID="FLATLIST"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`Credential test cancel sharing 2`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "marginRight": 5,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="CANCEL_BTN"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "fontSize": 17,
              "marginHorizontal": 10,
            },
            Object {
              "color": "#FFFFFF",
            },
            undefined,
          ]
        }
      >
        Cancel
      </Text>
    </View>
  </View>
</View>
`;

exports[`Credential test done button 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#000000",
      "borderRadius": undefined,
      "flex": 1,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <RCTScrollView
    contentOffset={
      Object {
        "x": NaN,
        "y": 0,
      }
    }
    data={
      Object {
        "claim": Object {
          "Serto ID": Object {
            "children": Array [
              Object {
                "age": 4,
                "name": "Bob",
              },
              Object {
                "age": 9,
                "name": "Alice",
              },
            ],
            "country": "USA",
            "dateOfBirth": "22-01-75",
            "name": "Sarah Adamson",
          },
        },
        "exp": 1579478400,
        "hash": "testHash",
        "iat": 1562769371,
        "iss": "Serto Identity Platform",
        "sub": "did:ethr:0xf3beac30c498d9e26865f34fcaa57dbb935b0d74",
        "type": "Some VC",
        "vc": Array [],
      }
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    snapToAlignment="center"
    stickyHeaderIndices={Array []}
    testID="FLATLIST"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`Credential test done button 2`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "marginRight": 5,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="DONE_BTN"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "fontSize": 17,
              "marginHorizontal": 10,
            },
            Object {
              "color": "#FFFFFF",
            },
            undefined,
          ]
        }
      >
        Done
      </Text>
    </View>
  </View>
</View>
`;

exports[`Credential test sharing button 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#000000",
      "borderRadius": undefined,
      "flex": 1,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <RCTScrollView
    contentOffset={
      Object {
        "x": NaN,
        "y": 0,
      }
    }
    data={
      Object {
        "claim": Object {
          "Serto ID": Object {
            "children": Array [
              Object {
                "age": 4,
                "name": "Bob",
              },
              Object {
                "age": 9,
                "name": "Alice",
              },
            ],
            "country": "USA",
            "dateOfBirth": "22-01-75",
            "name": "Sarah Adamson",
          },
        },
        "exp": 1579478400,
        "hash": "testHash",
        "iat": 1562769371,
        "iss": "Serto Identity Platform",
        "sub": "did:ethr:0xf3beac30c498d9e26865f34fcaa57dbb935b0d74",
        "type": "Some VC",
        "vc": Array [],
      }
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    snapToAlignment="center"
    stickyHeaderIndices={Array []}
    testID="FLATLIST"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`Credential test sharing button 2`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "marginRight": 5,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="SHARE_BTN"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "fontSize": 17,
              "marginHorizontal": 10,
            },
            Object {
              "color": "#FFFFFF",
            },
            undefined,
          ]
        }
      >
        Share
      </Text>
    </View>
  </View>
</View>
`;
